### ISSUE TITLE
"no repositories" view in list lacks ability to quickly add a repository

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#6474]
**Please describe the problem you think should be solved**

In #5878 we added the ability to quickly add a new repository when looking at the repository list:

![](https://user-images.githubusercontent.com/634063/46813533-2cba2880-cd77-11e8-96e8-77caac7a6586.png)

https://github.com/desktop/desktop/blob/3295ad860e746213db5ff74f97fbec40dc3041d9/app/src/ui/repositories-list/repositories-list.tsx#L170-L180

One limitation of this feature is that it's baked into the `FilterList`:

https://github.com/desktop/desktop/blob/3295ad860e746213db5ff74f97fbec40dc3041d9/app/src/ui/repositories-list/repositories-list.tsx#L151-L159

This is only rendered when there are repositories to display:

https://github.com/desktop/desktop/blob/3295ad860e746213db5ff74f97fbec40dc3041d9/app/src/ui/repositories-list/repositories-list.tsx#L123-L126

This means our "no repositories" view lacks the ability to quickly add repositories, so users will have to go back to the old way of opening the File menu to find these options:

<img width="305" src="https://user-images.githubusercontent.com/359239/50347915-aa37ad00-050d-11e9-9f9b-685eb3e243f5.png">

**[Optional] Do you have any potential solutions in mind?**

I'm not sure if this came up as part of the research behind the blank slate changes coming in 1.6, so cc @niik for any extra context before we start thinking about solutions

**Comment by niik** [comment]
> I'm not sure if this came up as part of the research behind the blank slate changes coming in 1.6, so cc @niik for any extra context before we start thinking about solutions

Yeah, this should be part of the onboarding work. I've already solved a very similar case for the clone filter list. I can take this and apply the same solution here.

**Comment by niik** [comment]
> This means our "no repositories" view lacks the ability to quickly add repositories, so users will have to go back to the old way of opening the File menu to find these options:

Reading this again, this is actually already fixed by #6365 since the repository sidebar won't be visible if there are no repositories. I'll add a `closes` to that PR.
